client/src/pages/Product.js

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from "react";
+import { Container, Grid, AppBar, Tabs, Tab, Button } from "@material-ui/core";
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import ImageGallery from "react-image-gallery";
+import "react-image-gallery/styles/css/image-gallery.css";
+
+function TabPanel(props) {
+    const { title, description, value, index, ...other } = props;
+    const classes = useStyles();
+
+    return (
+        <div
+            role="tabpanel"
+            hidden={value !== index}
+            id={`simple-tabpanel-${index}`}
+            aria-labelledby={`simple-tab-${index}`}
+            {...other}
+            className={classes.rootTab}
+        >
+            <h1>{title}</h1>
+            <p>{description}</p>
+        </div>
+    );
+}
+
+TabPanel.propTypes = {
+    children: PropTypes.node,
+    index: PropTypes.any.isRequired,
+    value: PropTypes.any.isRequired,
+};
+
+function a11yProps(index) {
+    return {
+        id: `simple-tab-${index}`,
+        'aria-controls': `simple-tabpanel-${index}`,
+    };
+}
+
+const useStyles = makeStyles((theme) => ({
+    rootImg: {
+        height:"700px",
+    },
+    rootGrid: {
+        marginTop: theme.spacing(6),
+    },
+    root: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.paper,
+    },
+    rootAppBar: {
+        backgroundColor: "#FFF",
+        color: '#000',
+        marginTop: theme.spacing(6),
+    },
+    rootTab: {
+        padding: "20px",
+    }
+}));
+
+export default function Product(props) {
+    const { params } = props.match;
+    const classes = useStyles();
+    const [value, setValue] = useState(0);
+    const [title, setTitle] = useState("");
+    const [description, setDescription] = useState("");
+    const [price, setPrice] = useState(0);
+    const [images, setImages] = useState([]);
+
+    useEffect(() => {
+        fetch("/product/" + params.id, {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            }
+        })
+            .then(res => {
+                return res.json();
+            })
+            .then(productJSON => {
+                setTitle(productJSON.title);
+                setDescription(productJSON.description);
+                setPrice(productJSON.price);
+                productJSON.photos.forEach(photo => {
+                    images.push({"original": photo, "thumbnail": photo});
```

Comment:
this may work because of the previous states you're setting but in general you want to avoid mutating an array in React because it does batch updates and may cause bugs at times. instead, for best practices you want to create a deep copy of this before you update the state for example

`setImages(productJSON.photos.map(photo => {"original": photo, "thumbnail": photo}))`